Userland/unzip.cpp

```diff
@@ -0,0 +1,132 @@
+{
+    u8 buffer[4];
+    off_t start_index = current_index == 0 ? current_index : current_index + 1;
+    for (off_t index = start_index; index < file_size - 4; index++) {
+        file->seek(index);
+        file->read(buffer, 4);
+        if (buffer[0] == central_directory_file_header_sig[0] && !memcmp(buffer, central_directory_file_header_sig, 4)) {
+            *return_index = index;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool unpack_file_for_central_directory_index(off_t central_directory_index, Core::File* file)
+{
+    u8 buffer[4];
+
+    off_t local_file_header_index;
+    file->seek(central_directory_index + 42);
+    file->read(buffer, 4);
+    local_file_header_index = buffer[3] << 24 | buffer[2] << 16 | buffer[1] << 8 | buffer[0];
+
+    off_t compressed_file_size;
+    file->seek(local_file_header_index + 18);
+    file->read(buffer, 4);
+    compressed_file_size = buffer[3] << 24 | buffer[2] << 16 | buffer[1] << 8 | buffer[0];
+
+    off_t file_name_length;
+    file->seek(local_file_header_index + 26);
+    file->read(buffer, 2);
+    file_name_length = buffer[1] << 8 | buffer[0];
+
+    off_t extra_field_length;
+    file->seek(local_file_header_index + 28);
+    file->read(buffer, 2);
+    extra_field_length = buffer[1] << 8 | buffer[0];
+
+    char file_name[file_name_length + 1];
+    file->seek(local_file_header_index + 30);
+    file->read((u8*)file_name, file_name_length);
+    file_name[file_name_length] = '\0';
+
+    auto new_file = Core::File::construct(String { file_name });
+    new_file->open(Core::IODevice::WriteOnly);
+
+    printf(" extracting: %s\n", file_name);
+    u8 file_contents[compressed_file_size];
+    file->seek(local_file_header_index + 30 + file_name_length + extra_field_length);
+    file->read(file_contents, compressed_file_size);
+
+    // FIXME: Try to uncompress data here. We're just ignoring it as no decompression methods are implemented yet.
+    new_file->write(file_contents, compressed_file_size);
+    new_file->close();
+    return true;
+}
+
+int main(int argc, char** argv)
+{
+    Vector<const char*> paths;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(paths, "File to unzip", "path", Core::ArgsParser::Required::Yes);
+    args_parser.parse(argc, argv);
+
+    if (paths.size() > 1) {
+        printf("Too many arguments\n");
+        return 1;
+    }
+
+    String zip_file_path = { paths[0] };
+
+    auto file = Core::File::construct(zip_file_path);
+    if (!file->open(Core::IODevice::ReadOnly)) {
+        printf("Cant read file %s. %s\n", zip_file_path.characters(), strerror(errno));
+        return 2;
+    }
+
+    struct stat stat;
+    int rc = lstat(zip_file_path.characters(), &stat);
+    if (rc < 0) {
+        perror("lstat");
+        memset(&stat, 0, sizeof(stat));
+    }
+
+    printf("Archive: %s\n", zip_file_path.characters());
```

Comment:
This (and the rest of logging) should go to `stderr`, shouldn't it?

Reply:
Now only the error text goes on stderr. I checked on my linux unzip command and the rest goes on stdout. So mimicking that